bootstrapper/cmd/bootstrapper/main.go

```diff
@@ -125,6 +125,8 @@ func main() {
 		if err != nil {
 			log.With(zap.Error(err)).Fatalf("Failed to set up cloud logger")
 		}
+		go metadata.PrepareAzureControlPlaneNode(ctx, log)
```

Comment:
Why is this done as a Go routine?
What would prevent us from simply executing it and waiting for an error?
The way this works right now could cause it to fail when the bootstrapper is done quickly and shuts down before the routine finishes

Reply:
Rewrote it as blocking function.